### ISSUE TITLE
RBMap::pop()

### ISSUE THREAD

**Issue by eadf** [dbyr/rb_tree#2]
A priority queue with `pop()` _and_ random access would make my Rust experiments so much more complete.
I've been using your `RBMap` with a hack to simulate `pop()`, something like this:

```
if let Some(key) = rbmap.iter().next().map(|x| *x.0) {
    site_events.remove_entry(&key)
} else {
    None
}
```
But this is not very elegant, and it does two lookups. So a real `pop()` would be so much better.
Maybe I'm overlooking something, but I can't seem to find any similar functionality in `RBMap`.

**Comment by dbyr** [comment]
Thanks for using it!
You're right, that is quite a nasty work around. The reason I didn't include a pop functionality for the map initially is because it's not really "standard" for a map. However, I can see that it may be useful in certain situations, and it's quite an easy addition. I'll chuck it in there today.

**Comment by dbyr** [comment]
I've added peek and pop methods for the RBMap, available in v0.3.2.

**Comment by eadf** [comment]
Thank you. 
I also have to thank you for this library, it's very nice and I hope it will get used more.
